Clarify controlled access within modularity and distributability (#191)
myyoda/principles-paper: main.tex

In practice, module boundaries often align with repository boundaries, but a module may also be a published container, a registered dataset, or any independently versioned unit.
Modularity applies this principle to research objects, structuring them as assemblies of modules whose boundaries are explicit.
Module boundaries are also where license compatibility must be checked, since combining modules with incompatible terms can prevent the assembled research object from being lawfully redistributed.
In addition to licenses, data use agreements and other legal, ethical, institutional, or technical constraints may impose different conditions on access and redistribution across a research object.

> For example, ABCD Study data are available to approved researchers under a Data Use Certification but not for unrestricted redistribution, even when organized to be technically distributable.

Reviewer: I would not bother with neuro-specific example at this level/stage

```suggestion
```

Reviewer: having reviewed and seeing some examples in other sections, I think we could potentially bring one up but it should be a bit more specific, as to pointing to a specific case, like e.g. following https://open-brain-consent.readthedocs.io/en/stable/ultimate.html#two-access-types-version-some-data-shared-publicly-more-data-shared-to-approved-researchers shared some portion publicly while keeping more restricted access to portions of the data.  

Note that quite often it is also done via "partial data release" where people share some trimmed version of the dataset while having more data; hence it is somewhat solved via "Modularity" at the lowest level of a module -- a file.

Author: I went with two examples from other domains to demonstrate universal relevance.

> These conditions may be managed through separate modules or permissions on individual files or objects, depending on which is most practical.

Reviewer: I feel that relating to underlying technologies used make sense in this context since decisions on level to modularize or control depend on the tech usually:

```suggestion
These conditions may be managed through separate modules or permissions on individual files or objects, depending on which is most practical given the technologies used.
```

> For example, the U.S. Census Bureau releases disclosure-protected public-use data while reserving detailed microdata for secure research environments \citep{united_states_census_bureau_dissemination_2025}; GBIF guidance similarly recommends publishing generalized species locations while providing precise coordinates only under agreement \citep{chapman_current_2020}.

Reviewer: not sure if we should expand GBIF (Global Biodiversity Information Facility) for readers here.  Really interesting (makes sense but I have not thought about it before) use case of not revealing detailed information in case of animal research.

Overall -- I like this addition and give a blessing -- we can improve later (deabbreviate) if needed.

> These conditions may be managed through separate modules, alternate representations, or permissions on individual files or objects, depending on the access-control mechanisms supported by the underlying technologies.

Reviewer 2: All these mechanisms are valid, but I'd argue they aren't equally-good options. I'm not experienced working with private data, so grain of salt, but it seems to me that isolating restricted components in their own module handles the constraint once, at a boundary, rather than through per-object settings that have to be set correctly every time and are easy to miss. 

What I have in mind would be to add M.4 requirement along the lines of: 

"Components requiring restricted access SHOULD be isolated in their own modules, so that access is controlled at a single boundary rather than per component."

Author: I am trying to acknowledge the messy reality of real world use cases here and the ways in which a project's history or objectives can result in different strategies for tiered release. I suspect they could all be argued to be the best in certain contexts. With different tools, data, different downstream users, regulations etc.  it seems too hard to assess what option is best without specific details of the project that produced a research object.

Reviewer 2: maybe demote to MAY then. STAMPED needs to be flexible to cover the messy reality, but I think we can hint at the cleanest idioms, even when they dont apply universally. IMO "separation of concerns" encourages the direction here, but I could be convinced otherwise.

Reviewer 3: "alternate representations" is too ambiguous

"separate modules" is great

> What I have in mind would be to add M.4 requirement along the lines of:
> 
> "Components requiring restricted access SHOULD be isolated in their own modules, so that access is controlled at a single boundary rather than per component."

I like it, and ought to be SHOULD level

Think in terms of DANDI, we already do this:
- Embargoed Dandisets are restricted at the module level; only certain individuals can access their components, and when they can, they can access ALL of the components of the module

Now, people have requested per-asset read permissions for 'restricted content', for which there are even design docs, but the fact it hasn't been done in 4+ years reflects how much messier it is to do it that way instead of at the module level, reflecting how module level control really is the cleanest implementation of any such feature. Even with a per-asset feature like that, you will inevitably have more components than modules, and the chance of a mistake in permission setting per component thus drives the total probability of a single mistake higher than that of a single mistake at a module level

Reviewer 2: "Alternate representations" agreed too ambiguous, I took it to mean outputs, ie a public output and a private output, and even in that case I imagine those would be separate modules as well.

Reviewer: IMHO it is worth proposing specific edits at this point to make this discussion productive. Remember that here we are talking about Modularization as solution for varying accessibility.  I would just propose to

```suggestion
These conditions MAY be managed through separate modules, altered (e.g., encrypted or anonymized) representations, or permissions on individual files or objects, depending on the access-control mechanisms supported by the underlying technologies.
```

and move on to not make "perfect be the enemy of the good"

Reviewer: note: this discussion is over an already outdated fragment.

Though persistent retrievability is a necessary but not sufficient condition for redistribution.
Additionally, each module should carry an easily identifiable license or set of licenses for any bundled assets (e.g., via SPDX identifiers\cite{stewart_software_2010} or the REUSE specification\cite{fsfe_reuse_2024}) so that downstream researchers can compose, modify, and re-share components without legal ambiguity.

> The spectrum of Distributability~(D) intersects with FAIR Accessibility: whether access is public or restricted, STAMPED addresses operational aspects of distribution, such as persistent packaging and documented retrieval.

Reviewer 2: I think this sentence waters down the goal of this "spectrum minimum" sentence. I prefer your phrasing "intersects with" FAIR, but "STAMPED addresses..." is circular, this is where we should establish the minimum. 

```suggestion
The spectrum of Distributability~(D) begins at its intersection with FAIR Accessibility: components are accessible and retrieval instructions are documented. 
This holds even where access to some components is restricted; the research object documents how each component is retrieved, while granting access remains the responsibility of its provider.
```

Author: > This holds even where access to some components is restricted

I think tiered access or documentation  of retrieval (gated by provider granting access) are part of the intersection. I was trying to avoid implying otherwise. There's plenty that STAMPED adds to that; we just need to express the added value as concisely as possible given the current flow... or consider moving the detail elsewhere.

Reviewer 2: Maybe we could keep 300 short, and then 301 can go into more detail about the intersection with FAIR?

> The spectrum of distributability intersects with FAIR Accessibility: components may be publicly retrievable or subject to authentication and authorization, while STAMPED additionally asks whether they are packaged and maintained in a persistent state that can be distributed when permission allows.

Reviewer: I feel that we want smth in the middle of two phrasings so we do not over specify details, but also I think Distributability is not limited to "Public" -- it is overall a matter of being able to distribute (publicly or privately) ... may be smth like

```suggestion
The spectrum of distributability intersects with FAIR Accessibility, where STAMPED operationalizes distribution relates aspects, e.g. via documented retrieval instructions, which would improve accessibility.
```

Author: I went with:

The spectrum of Distributability~(D) intersects with FAIR Accessibility: whether access is public or restricted, STAMPED addresses operational aspects of distribution, such as persistent packaging and documented retrieval.
